### PR TITLE
Add a background janitor thread to the disk cache.

### PR DESCRIPTION
**Pull request by tylerwilliams**
This should prevent the cache from ever reaching max capacity, which means that writes will not have to evict something before completing.

### CODE REVIEW

File: server/backends/disk_cache/disk_cache_test.go [hunk 1]
```diff
@@ -354,3 +354,42 @@ func TestAsyncLoading(t *testing.T) {
 		}
 	}
 }
+
+func TestJanitorThread(t *testing.T) {
+	maxSizeBytes := int64(10000000) // 10MB
```
Comment (vadimberezniker) [inline server/backends/disk_cache/disk_cache_test.go:359]: nit: use digit separators (_) to make long number literals easier to read

File: server/backends/disk_cache/disk_cache_test.go [hunk 2]
```diff
@@ -354,3 +354,42 @@ func TestAsyncLoading(t *testing.T) {
...
+
+	// Make a new disk cache with a smaller size. The
+	// janitor should clean extra data up, oldest first.
+	dc, err = disk_cache.NewDiskCache(rootDir, maxSizeBytes/2) // 5MB
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(time.Second)
+	for i, d := range digests {
+		if i < 500 {
```
Comment (vadimberezniker) [inline server/backends/disk_cache/disk_cache_test.go:385]: you can avoid the indent if you change this to 
if i >= 500 {
  break
}

File: server/backends/disk_cache/disk_cache_test.go [hunk 3]
```diff
@@ -354,3 +354,42 @@ func TestAsyncLoading(t *testing.T) {
...
+		digests = append(digests, d)
+	}
+
+	// Make a new disk cache with a smaller size. The
+	// janitor should clean extra data up, oldest first.
+	dc, err = disk_cache.NewDiskCache(rootDir, maxSizeBytes/2) // 5MB
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(time.Second)
```
Comment (vadimberezniker) [inline server/backends/disk_cache/disk_cache_test.go:383]: nit: instead of sleeping, repeat the check until it returns true or timeout is reached

**Review by vadimberezniker — APPROVED** [review]
LGTM

File: server/backends/disk_cache/disk_cache_test.go (hunk 3 above)
Comment (tylerwilliams) [reply server/backends/disk_cache/disk_cache_test.go]: I tried to make this work but it's complicated by the fact that checking the cache bumps the record's TTL. So for now I've just reduced the sleep to a short but manageable time and we can revisit if this becomes an issue.

**Pull request merged**